### PR TITLE
add new help link to safe browsing android

### PR DESCRIPTION
**Pull request by tapanmodh**
Resolves https://github.com/brave/brave-browser/issues/20514

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by alexsafe — COMMENTED** [review]
@tapanmodh I think you need to run `git cl format` or `npm runt lint` . The CI job seems to be failing at lint checks

File: android/java/org/chromium/chrome/browser/help/BraveHelpAndFeedbackLauncher.java [hunk 1]
```diff
@@ -12,20 +12,37 @@
...
+            launchFallbackSupportUri(activity);
+        }
+    }
+
+    private void launchSafeBrowsingUri(Context context) {
+
+        Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(SAFE_BROWSING_URL));
+        // Let Chrome know that this intent is from Chrome, so that it does not close the app when
+        // the user presses 'back' button.
+        intent.putExtra(Browser.EXTRA_APPLICATION_ID, context.getPackageName());
```
[the remark this answers is not in the document]
Comment (tapanmodh) [reply android/java/org/chromium/chrome/browser/help/BraveHelpAndFeedbackLauncher.java:43]: yes, it is working on android 12

[the remark this answers is not in the document]
Comment (SergeyZhukovsky) [reply android/java/org/chromium/chrome/browser/help/BraveHelpAndFeedbackLauncher.java:43]: I believe `FLAG_IMMUTABLE` on Android 12 is related to a `PendingIntent` only

File: android/java/org/chromium/chrome/browser/help/BraveHelpAndFeedbackLauncher.java [hunk 2]
```diff
@@ -12,20 +12,38 @@
...
+                    activity.getResources().getString(R.string.help_context_safe_browsing))) {
+            launchSafeBrowsingUri(activity);
+        } else {
+            launchFallbackSupportUri(activity);
+        }
+    }
+
+    private void launchSafeBrowsingUri(Context context) {
+        Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(SAFE_BROWSING_URL));
+        // Let Chrome know that this intent is from Chrome, so that it does not close the app when
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/help/BraveHelpAndFeedbackLauncher.java:41]: I would change Chrome to Brave

File: android/java/org/chromium/chrome/browser/help/BraveHelpAndFeedbackLauncher.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (tapanmodh) [reply android/java/org/chromium/chrome/browser/help/BraveHelpAndFeedbackLauncher.java:43]: @SergeyZhukovsky Yes

**Review by deeppandya — APPROVED** [review]
LGTM

File: android/java/org/chromium/chrome/browser/help/BraveHelpAndFeedbackLauncher.java (hunk 2 above)
Comment (tapanmodh) [reply android/java/org/chromium/chrome/browser/help/BraveHelpAndFeedbackLauncher.java]: done

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Pull request merged**